gadopt/gplates.py

```diff
@@ -0,0 +1,449 @@
+            )
+        )
+
+        # last reconstruction time
+        self.reconstruction_time = None
+
+        # Flag to know when to recalculate surface velocities.
+        self.recalculation_flg = False
+
+        # Factor to scale plate velocities to RMS velocity of model,
+        # the definition: RMS_Earth / RMS_Model is used
+        # This is specially used in low-Rayleigh-number simulations
+        self.scaling_factor = scaling_factor
+
+    # setting the time that we are interested in
+    def get_plate_velocities(self, target_coords, model_time):
+        """
+        Returns plate velocities for the specified target coordinates at the top boundary of a sphere,
+        for a given model time, by integrating plate tectonic reconstructions from pyGplates.
+
+        This method calculates new plate velocities.
+        It utilizes the cKDTree data structure for efficient nearest neighbor
+        searches to interpolate velocities onto the mesh nodes.
+
+        Args:
+            target_coords (array-like): Coordinates of the points at the top of the sphere.
+            model_time (float): The model time for which plate velocities are to be calculated and assigned.
+                This time is converted to geological time and used to extract relevant plate motions
+                from the pyGplates model.
+
+        Raises:
+            Exception: If the requested reconstruction time is geologically negative, indicating an issue with
+                the time conversion.
+
+        Notes:
+            - The method uses conversions between non-dimensionalised time and geologic time.
+            - Velocities are non-dimensionalised and scaled for the simulation before being applied.
+        """
+
+        # Raising an error if the user is asking for invalid time
+        if self.ndtime2geotime(ndtime=model_time) < 0:
+            raise Exception(
+                ("pyGplates: geologic time is being negative!"
+                 f"maximum: {self.geologic_zero/(pyGplatesConnector.time_dim_factor/pyGplatesConnector.myrs2sec/self.scaling_factor)}")
+            )
+
+        # compute new velocities
+        self.reconstruction_time = self.ndtime2geotime(model_time)
+        self.interpolated_u = self._interpolate_seeds_u(target_coords)
+        return self.interpolated_u
+
+    def ndtime2geotime(self, ndtime):
+        """
+        Converts non-dimensionalized time to geologic time with respect to present-day (Ma).
+
+        This function takes a non-dimensionalized time value and converts it to geologic time,
+        measured in millions of years ago (Ma), with respect to the present day.
+
+        Args:
+            ndtime (float): The non-dimensionalized time to be converted.
+
+        Returns:
+            float: The converted geologic time in millions of years ago (Ma).
+        """
+
+        geotime = self.geologic_zero - float(ndtime) * pyGplatesConnector.time_dim_factor/pyGplatesConnector.myrs2sec/self.scaling_factor
```

Comment:
You said `ndtime` should be a float in the function signature already? Yes, that's not enforced, but I think it could introduce subtle bugs if somebody ends up relying on the implicit conversion.

```suggestion
        geotime = self.geologic_zero - ndtime * pyGplatesConnector.time_dim_factor/pyGplatesConnector.myrs2sec/self.scaling_factor
```

Also, you end up using `pyGplatesConnector.time_dim_factor/pyGplatesConnector.myrs2sec/self.scaling_factor` a few times, maybe that should get its own constant?